typewise-alert.cpp

```diff
@@ -1,71 +1,52 @@
+#define ENABLE_BREACH_STATUS
+
 #include "typewise-alert.h"
-#include <stdio.h>
 
 BreachType inferBreach(double value, double lowerLimit, double upperLimit) {
+
+  BreachType retBreachType = NORMAL;
   if(value < lowerLimit) {
-    return TOO_LOW;
+    retBreachType = TOO_LOW;
   }
   if(value > upperLimit) {
-    return TOO_HIGH;
+    retBreachType = TOO_HIGH;
   }
-  return NORMAL;
+  return retBreachType;
 }
 
-BreachType classifyTemperatureBreach(
-    CoolingType coolingType, double temperatureInC) {
-  int lowerLimit = 0;
-  int upperLimit = 0;
-  switch(coolingType) {
-    case PASSIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 35;
-      break;
-    case HI_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 45;
-      break;
-    case MED_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 40;
-      break;
-  }
-  return inferBreach(temperatureInC, lowerLimit, upperLimit);
+void classifyTemperatureBreach(std::unordered_map <int, std::pair<int, int>> &tempBreach) {
+
+      tempBreach.insert({PASSIVE_COOLING, std::make_pair<int, int>(0, 35)});
+      tempBreach.insert({HI_ACTIVE_COOLING, std::make_pair<int, int>(0, 45)});
+      tempBreach.insert({MED_ACTIVE_COOLING, std::make_pair<int, int>(0, 40)});
+      
 }
 
 void checkAndAlert(
     AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC) {
 
-  BreachType breachType = classifyTemperatureBreach(
-    batteryChar.coolingType, temperatureInC
-  );
+    std::unordered_map <int, std::pair<int, int>> temp_Breach;
+  classifyTemperatureBreach(temp_Breach);
 
-  switch(alertTarget) {
-    case TO_CONTROLLER:
-      sendToController(breachType);
-      break;
-    case TO_EMAIL:
-      sendToEmail(breachType);
-      break;
-  }
-}
+  int lowerLimit, upperLimit;
+
+  // find cooling type and update lower&upper limit
+  auto itr = temp_Breach.find(batteryChar.coolingType);
+  lowerLimit = itr->second.first;
+  upperLimit = itr->second.second;
+  BreachType breachType = inferBreach(temperatureInC, lowerLimit, upperLimit);
+  sendBatteryTempStatus(alertTarget, breachType);
 
-void sendToController(BreachType breachType) {
-  const unsigned short header = 0xfeed;
-  printf("%x : %x\n", header, breachType);
 }
 
-void sendToEmail(BreachType breachType) {
-  const char* recepient = "a.b@c.com";
-  switch(breachType) {
-    case TOO_LOW:
-      printf("To: %s\n", recepient);
-      printf("Hi, the temperature is too low\n");
-      break;
-    case TOO_HIGH:
-      printf("To: %s\n", recepient);
-      printf("Hi, the temperature is too high\n");
-      break;
-    case NORMAL:
-      break;
+void sendBatteryTempStatus(AlertTarget alertTarget, BreachType breachType, std::ostream& os, const char* recepientId) {
```

Comment:
There is no call from production code with both alertTarget as email and recipient id set.
Please modify the code and add a test that has alert target as TO_EMAIL and recipient ID, or else the hardcoded email id will be used in the production envt.